### ISSUE TITLE
Создание и обработка связного списка и сохраниние его в файл

### ISSUE THREAD

**Issue by Geniok** [Pahtus200Maks0/Test#3]
Здесь основное замечание, это использование глобальных указателей

> struct order *first;
struct order *current;
struct order *newNode;

Это плохая практика, использовать такие глобальные указатели. 

Еще небольшое замечание, при работе со списками имеются общепринятые наименования функции  push(), pop() и так далее, лучше использовать их при именовании своих функций.

Еще такой вопрос, ваш список хранит 

> int number;
    int sum;

Не могли бы вы пояснить, что эти данные означают ?

Не совсем ясно предназначение функции deletnode() и check().
Не могли бы вы пояснить. зачем они нужны.

**Comment by Pahtus200Maks0** [comment]
1 - основное замечание, это использование глобальных указателей - я так понимаю лучше использовать указатели с ключевым словом static в пределах main()?. Чтоб область видимости захватила весь файл? 

2 - int number; - номер счёта
     int sum; - сумма счёта 

3 - check() - если номер счёта начинает повторятся, она выявляет это и не даёт запустить цикл дальше, а сразу вызывает функцию deletnode() - которая убирает память выделенную под повторный узел. И после цикл повторяется с соответствующим уведомлением

**Comment by Geniok** [comment]
> 1 - основное замечание, это использование глобальных указателей - я так понимаю лучше использовать указатели с ключевым словом static в пределах main()?. Чтоб область видимости захватила весь файл?

Нет, неверно. Вам просто стоит объявлять новый указатель тогда, когда он нужен т в том месте, где он нужен. Указатель не занимает память.


> 
> 2 - int number; - номер счёта int sum; - сумма счёта

Я почему спросил, в своей программе вы запрашиваете не номер, а индекс. Индекс обычно определяет порядковый номер, когда числа идут друг за другом в порядке возрастания, то есть 1, 2, 3, 4, 5. Меня это несколько смутило и стало любопытно, что же вы все таки запрашиваете у пользователя, индекс или номер счета.
